src/scripts/specialEvents/SpecialEvent.ts

```diff
@@ -78,39 +111,33 @@ class SpecialEvent {
         }, Math.max(0, timeTillEventStart));
     }
 
-    shouldStartNow(): boolean {
-        // If passed the start time, and not ended already
-        return this.timeTillStart() <= 0 && !this.hasEnded();
-    }
-
-    timeTillStart(): number {
-        return +this.startTime - Date.now();
-    }
-
-    timeTillEnd(): number {
-        return +this.endTime - Date.now();
-    }
-
-    hasStarted(): boolean {
-        return this.status == SpecialEventStatus.started;
-    }
+    start() {
+        // Update event status
+        this.status = SpecialEventStatus.started;
 
-    hasEnded(): boolean {
-        return this.status == SpecialEventStatus.ended;
-    }
+        // We only wan't the notification displayed for 1 hour, or until the event is over
+        const timeTillEventEnd = this.timeTillEnd();
+        this.notify('on now!', Math.min(1 * GameConstants.HOUR, timeTillEventEnd), GameConstants.NotificationOption.success);
 
-    notify(time: string, timeout: number, type = GameConstants.NotificationOption.info) {
-        timeout -= 1000;
-        Notifier.notify({ title: `[EVENT] ${this.title}`, message: `${this.description}<br/><br/><strong>Start time:</strong> ${this.startTime.toLocaleString()}<br/><strong>End time:</strong> ${this.endTime.toLocaleString()}`, type, time, timeout, setting: GameConstants.NotificationSetting.event_start_end });
+        this.startFunction();
+        // Start checking when the event should be ending
+        this.checkEnd();
     }
 
-    start() {
-        this.status = SpecialEventStatus.started;
+    checkEnd() {
         const timeTillEventEnd = this.timeTillEnd();
 
-        this.notify('on now!', Math.min(1 * GameConstants.HOUR, timeTillEventEnd), GameConstants.NotificationOption.success);
-        this.startFunction();
+        // If more than 1 day, don't notify the player yet
+        if (timeTillEventEnd > 1 * GameConstants.DAY) {
+            // Check again when 7 days left until event start, or in 7 days (whichever is less)
```

Comment:
This comment doesn't match the code below :eyes:

Reply:
```suggestion
            // Check again when 1 day left until event start, or in 1 day (whichever is less)
```

Reply:
Whoops 😅